src/app/sign-in/pages/sign-in/sign-in.component.ts

```diff
@@ -1,13 +1,139 @@
-import { Component, OnInit } from '@angular/core'
-import { MatDialog } from '@angular/material/dialog'
+import { Component, Inject, OnInit } from '@angular/core'
+import { FormControl, FormGroup, Validators } from '@angular/forms'
+import { WINDOW } from '../../../cdk/window'
+import { SignInService } from '../../../core/sign-in/sign-in.service'
+import { UserService } from '../../../core'
+import { environment } from 'src/environments/environment'
+import { TwoFactorComponent } from '../../components/two-factor/two-factor.component'
+import { take } from 'rxjs/operators'
 
 @Component({
   selector: 'app-sign-in',
   templateUrl: './sign-in.component.html',
-  styleUrls: ['./sign-in.component.scss'],
+  styleUrls: [
+    './sign-in.component.scss-theme.scss',
+    './sign-in.component.scss',
+  ],
+  providers: [TwoFactorComponent],
 })
 export class SignInComponent implements OnInit {
-  constructor(private dialog: MatDialog) {}
+  loading = false
+  badCredentials = false
+  printError = false
+  showDeprecatedError = false
+  showDeactivatedError = false
+  showUnclaimedError = false
+  showBadVerificationCode = false
+  showBadRecoveryCode = false
+  showInvalidUser = false
+  show2FA = false
+  isLoggedIn = false
+  displayName: string
+  realUserOrcid: string
+  email: string
+  orcidPrimaryDeprecated: string
+
+  constructor(
+    private _signIn: SignInService,
+    private _userInfo: UserService,
+    @Inject(WINDOW) private window: Window
+  ) {
+    _userInfo
+      .getUserStatus()
+      .pipe(take(1))
+      .subscribe(data => {
+        if (data) {
+          this.isLoggedIn = data
+          _userInfo
+            .getUserInfoOnEachStatusUpdate()
+            .pipe(take(1))
+            .subscribe(info => {
+              this.displayName = info.displayName
+              this.realUserOrcid =
+                environment.BASE_URL + info.userInfo.REAL_USER_ORCID
+            })
+        }
+      })
+  }
+
+  usernameFormControl = new FormControl('', [
+    Validators.required,
+    Validators.pattern(
+      '(\\d{4}-){3}\\d{3}[\\dX]' +
+        '|^.*\\.([a-zA-Z\\-])([a-zA-Z\\-]{0,61})([a-zA-Z\\-])$'
+    ),
+  ])
```

Comment:
@DanielPalafox I think I explain wrong with this one https://github.com/ORCID/orcid-angular/pull/148#discussion_r408546091

We do need to create a new custom validator, to validate "orcidOrEmail" 

Something like

```
  static orcidOrEmail(pattern: string | RegExp): ValidatorFn {
    return (control: AbstractControl) => {

      If (Validators.email(control) || Validators.pattern(TLD_REGEXP)(control) && Validators.pattern(ORCID_REGEXP)(control) ){
    return {'invalidUserName': true}
    }
    return null
    }
  }
```

After that use that validator on the usernameFormControl  like

```
usernameFormControl = new FormControl('', [
    Validators.required,
    this.isEmailOrPassword()
```

The recent change to this code only uses the TLD_REGEXP for emails but that only works well in combination with the Angular email validator. So this custom validator should solve that

Note:  please import constants from the constants.ts file when those are required.